src/ui/Combobox/ComboboxValues.ts

```diff
@@ -171,20 +202,47 @@ export class ComboboxValues {
 
   private get nextOrFirstValue() {
     if (!this.keyboardActiveValue) {
-      return this.values[0];
+      return this.firstValue;
     }
 
     const nextValueIndex = (this.values.indexOf(this.keyboardActiveValue) + 1) % this.values.length;
     return this.values[nextValueIndex];
   }
 
+  private get firstValue() {
+    return this.values[0];
+  }
+
   private get previousOrLastValue() {
-    const lastValueIndex = this.values.length - 1;
     if (!this.keyboardActiveValue) {
-      return this.values[lastValueIndex];
+      return this.lastValue;
     }
 
     const previousValueIndex = this.values.indexOf(this.keyboardActiveValue) - 1;
-    return previousValueIndex >= 0 ? this.values[previousValueIndex] : this.values[lastValueIndex];
+    return previousValueIndex >= 0 ? this.values[previousValueIndex] : this.values[this.values.length - 1];
```

Comment:
this could be shortened to:
```ts
private get previousOrLastValue() {
  if (!this.keyboardActiveValue) {
    return this.lastValue;
  }
  const activeIndex = this.values.indexOf(this.keyboardActiveValue);
  return activeIndex > 0 ? this.values[activeIndex - 1] : this.lastValue;
  // if `activeIndex` cannot be `-1`, then then `activeIndex > 0` could be shortened to just `activeIndex`.
}
```